lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,11 +20,13 @@ def as_json
 
         def type_for(serializer)
           return serializer._type if serializer._type
-          if ActiveModelSerializers.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
-          else
-            serializer.object.class.model_name.plural
+          type = serializer.object.class.to_s.split('::')
+          type.map! { |t| t.underscore.downcase }
```

Comment:
Yeah, as I mentioned in PR description, type name transform should be customizable too, but I'm not sure if that's always the case that it's the same as `key_transform`. Is it? Technically we could apply `key_transform` on type's segments. That would be pretty easy, but is it flexible enough?